marklogic-data-hub/src/main/resources/hub-internal-config/security/roles/data-hub-common-writer.json

```diff
@@ -29,6 +29,11 @@
       "privilege-name": "rest-writer",
       "action": "http://marklogic.com/xdmp/privileges/rest-writer",
       "kind": "execute"
+    },
+    {
+      "privilege-name": "xdmp:invoke",
```

Comment:
One downside to JSON is we don't have a way to have inline comments about why a privilege is needed. I've used README.md files in the same directory as JSON files before to provide those comments. But those can be difficult to keep in sync - someone has to remember to update them. 

In theory though, if someone wonders why this privilege is set, they could remove it and run the test suite and something should fail, which would point out what code needs the privilege, right?

Reply:
True. This one stems from the need to insert via hub-utils. It was missed when doing the mapping roles since the mapping reader already provided it as part of testing the mapping.